adagrams/game.py

```diff
@@ -1,11 +1,105 @@
+import random
+
 def draw_letters():
-    pass
+    hand = []
+    letters_list = []
+    LETTER_POOL = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3,
+    'H': 2, 'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6,
+    'O': 8, 'P': 2, 'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4,
+    'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+    }
+
+    #make dictionary into a list 
+    for letter, count in LETTER_POOL.items():
+        letters_list.extend(letter * count)
+
+    #make a hand of 10 letters 
+    
+    letters = []
+    temporary_letters_list = letters_list.copy()
+    for letter in range(0,10):
+        
+        random_number = random.randint(0,len(temporary_letters_list)-1)
+
+        #remove the letter from list 
+        random_letter = temporary_letters_list.pop(random_number)
+        letters.append(random_letter)
+    
+    return letters 
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    temporary_letter_bank = letter_bank[:]
+    for letter in word: 
+        if letter not in temporary_letter_bank:  
+            return False
+        temporary_letter_bank.remove(letter) 
+    return True
 
 def score_word(word):
-    pass
+
+    score_one_point = ["A","E","I","O","U","L","N","S","T"]
+    score_two_point =["D","G"]
+    score_three_point = ["B","C","M","P"]
+    score_four_point = ["F","H","V","W","Y"]
+    score_five_point = ["K"]
+    score_eight_point = ["J","X"]
+    score_ten_point = ["Q","Z"]
```

Comment:
How could we change the structure of this data so that we could make the loop that follows more concise? (Hint: think about a dictionary and it's methods)